pkg/api/stewardship.go

```diff
@@ -16,10 +19,31 @@ import (
 func (s *Service) stewardshipPutHandler(w http.ResponseWriter, r *http.Request) {
 	nameOrHex := mux.Vars(r)["address"]
 	address, err := s.resolveNameOrAddress(nameOrHex)
-	if err != nil {
+	switch {
+	case errors.Is(err, resolver.ErrParse):
 		s.logger.Debug("stewardship put: parse address string failed", "string", nameOrHex, "error", err)
 		s.logger.Error(nil, "stewardship put: parse address string failed")
-		jsonhttp.NotFound(w, nil)
+		jsonhttp.BadRequest(w, "invalid address")
+		return
+	case errors.Is(err, resolver.ErrNotFound):
+		s.logger.Debug("stewardship put: address not found", "string", nameOrHex, "error", err)
+		s.logger.Error(nil, "stewardship put: address not found")
+		jsonhttp.NotFound(w, "address not found")
+		return
+	case errors.Is(err, resolver.ErrServiceNotAvailable):
+		s.logger.Debug("stewardship put: service unavailable", "string", nameOrHex, "error", err)
+		s.logger.Error(nil, "stewardship put: service unavailable")
+		jsonhttp.ServiceUnavailable(w, nil)
+		return
+	case errors.Is(err, resolver.ErrInvalidContentHash):
+		s.logger.Debug("stewardship put: invalid content hash", "string", nameOrHex, "error", err)
+		s.logger.Error(nil, "stewardship put: invalid content hash")
+		jsonhttp.Respond(w, http.StatusUnprocessableEntity, "stewardship put: invalid content hash")
```

Comment:
Why do we use a quite obscure HTTP error number (422 Unprocessable Entity) here when there is a simple well-known alternative to it (400 Bad Request)?

As far as I understand we don't even use 422 correctly and that may be also because it's obscure.
> The 422 (Unprocessable Content) status code indicates that the server understands the content type of the request content (hence a [415 (Unsupported Media Type)](https://httpwg.org/specs/rfc9110.html#status.415) status code is inappropriate), and the syntax of the request content is correct, but it was unable to process the contained instructions. For example, this status code can be sent if an XML request content contains well-formed (i.e., syntactically correct), but semantically erroneous XML instructions.
https://httpwg.org/specs/rfc9110.html#status.422